diff --git a/includes/add-content.inc b/includes/add-content.inc
index 8878253..09dc765 100644
--- a/includes/add-content.inc
+++ b/includes/add-content.inc
@@ -43,6 +43,7 @@ function template_preprocess_panels_add_content_modal(&$vars) {
  */
 function template_process_panels_add_content_modal(&$vars) {
   $content = !empty($vars['categories'][$vars['category']]['content']) ? $vars['categories'][$vars['category']]['content'] : array();
+  $vars['columns'] = array();
 
   // If no category is selected or the category is empty or our special empty
   // category render a 'header' that will appear instead of the columns.
@@ -56,9 +57,7 @@ function template_process_panels_add_content_modal(&$vars) {
     // This will default to 2 columns in the theme definition but could
     // theoretically be changed by a preprocess.
     $columns = $vars['column_count'];
-    foreach (range($columns) as $column) {
-      $vars['columns'][$column] = '';
-    }
+    $vars['columns'] = array_fill(1, $columns, '');
 
     $col_size = count($titles) / $columns;
     $count = 0;
@@ -76,12 +75,10 @@ function template_process_panels_add_content_modal(&$vars) {
  */
 function template_preprocess_panels_add_content_link(&$vars) {
   $vars['title'] = filter_xss_admin($vars['content_type']['title']);
-  $vars['description'] = isset($vars['content_type']['description']) ? $vars['content_type']['description'] : $title;
+  $vars['description'] = isset($vars['content_type']['description']) ? $vars['content_type']['description'] : $vars['title'];
   $vars['icon'] = ctools_content_admin_icon($vars['content_type']);
   $vars['url'] = $vars['renderer']->get_url('add-pane', $vars['region'], $vars['content_type']['type_name'], $vars['content_type']['subtype_name']);
 
   $vars['image_button'] = ctools_ajax_image_button($vars['icon'], $vars['url'], $vars['description'], 'panels-modal-add-config');
   $vars['text_button'] = ctools_ajax_text_button($vars['title'], $vars['url'], $vars['description'], 'panels-modal-add-config');
-
-  error_log('test');
 }
